ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/checkpointsync/CheckpointDownloadBlockStep.java

```diff
@@ -65,17 +83,48 @@ public CompletableFuture<Optional<BlockWithReceipts>> downloadBlock(final Hash h
   private CompletableFuture<Optional<BlockWithReceipts>> downloadReceipts(
       final PeerTaskResult<Block> peerTaskResult) {
     final Block block = peerTaskResult.getResult();
-    final GetReceiptsFromPeerTask getReceiptsFromPeerTask =
-        GetReceiptsFromPeerTask.forHeaders(ethContext, List.of(block.getHeader()), metricsSystem);
-    return getReceiptsFromPeerTask
-        .run()
-        .thenCompose(
-            receiptTaskResult -> {
-              final Optional<List<TransactionReceipt>> transactionReceipts =
-                  Optional.ofNullable(receiptTaskResult.getResult().get(block.getHeader()));
-              return CompletableFuture.completedFuture(
-                  transactionReceipts.map(receipts -> new BlockWithReceipts(block, receipts)));
-            })
-        .exceptionally(throwable -> Optional.empty());
+    if (synchronizerConfiguration.isPeerTaskSystemEnabled()) {
+      CompletableFuture<Optional<BlockWithReceipts>> futureReceipts = new CompletableFuture<>();
+      GetReceiptsFromPeerTask task =
+          new GetReceiptsFromPeerTask(
+              List.of(block.getHeader()), new BodyValidator(), currentProtocolSpecSupplier);
+      PeerTaskExecutorResult<Map<BlockHeader, List<TransactionReceipt>>> executorResult =
+          peerTaskExecutor.execute(task);
+
+      if (executorResult.responseCode() == PeerTaskExecutorResponseCode.SUCCESS) {
+        List<TransactionReceipt> transactionReceipts =
+            executorResult
+                .result()
+                .map((map) -> map.get(block.getHeader()))
+                .orElseThrow(
+                    () ->
+                        new IllegalStateException("PeerTask response code was success, but empty"));
+        if (block.getBody().getTransactions().size() != transactionReceipts.size()) {
+          throw new IllegalStateException(
+              "PeerTask response code was success, but incorrect number of receipts returned");
+        }
```

Comment:
There is no need to recheck these these validations as these are already done in GetReceiptsFromPeerTask